src/core/lib/channel/channel_args.h

```diff
@@ -324,6 +324,18 @@ class ChannelArgs {
     const grpc_arg_pointer_vtable* vtable_;
   };
 
+  // Helper to create a `Pointer` object to an object that is not owned by the
+  // `ChannelArgs` object. Useful for tests, a code smell for production code.
+  template <typename T>
+  static Pointer UnownedPointer(T* p) {
```

Comment:
I'm not crazy about adding this method to this API, given that it's not something we actually want production code to use.  Wouldn't it be better to have a utility function in some test library to do this instead?